src/utils/expensesFilter.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const expensesService = require('../services/expenses.services');
+
+function isDateAfter(dateToCheck, referenceDate) {
+  const dateA = new Date(dateToCheck).valueOf();
+  const dateB = new Date(referenceDate).valueOf();
+
+  return dateA > dateB;
+}
+
+function isDateBefore(dateToCheck, referenceDate) {
+  const dateA = new Date(dateToCheck).valueOf();
+  const dateB = new Date(referenceDate).valueOf();
+
+  return dateA < dateB;
+}
+
+const expensesFilter = ({
+  userId,
+  from,
+  to,
+  categories,
+}) => {
+  let expenses = expensesService.getAll();
+
+  if (userId) {
+    expenses = expenses
+      .filter(expense => expense.userId === Number(userId));
+  }
+
+  if (categories) {
+    if (Array.isArray(categories)) {
+      expenses = expenses
+        .filter(({ category }) => categories.includes(category));
+    } else {
+      expenses = expenses
+        .filter(({ category }) => category === categories);
+    }
```

Comment:
Better not to nest `if`
Would be better to create variable and transform `categories` to array
And then just use `filter` without `if/else`